src/Entity/ListBuilder/SubscriptionListBuilder.php

```diff
@@ -248,10 +210,60 @@ public function buildRow(EntityInterface $entity) {
    * {@inheritdoc}
    */
   public function render() {
-    $build = parent::render();
+    $build = [];
+
+    // Group the subscriptions by status.
+    $subscriptions = [
+      SubscriptionInterface::STATUS_ACTIVE => [
+        'title' => $this->t('Active and Future Plans'),
+        'entities' => [],
+      ],
+      SubscriptionInterface::STATUS_ENDED => [
+        'title' => $this->t('Cancelled and Expired Plans'),
+        'entities' => [],
+      ],
+    ];
+
+    /** @var \Drupal\apigee_m10n\Entity\SubscriptionInterface $entity */
+    foreach ($this->load() as $entity) {
+      if (in_array($entity->getSubscriptionStatus(), [SubscriptionInterface::STATUS_ACTIVE, SubscriptionInterface::STATUS_FUTURE])) {
+        $subscriptions[SubscriptionInterface::STATUS_ACTIVE]['entities'][] = $entity;
+        continue;
+      }
 
-    array_push($build['table']['#cache']['contexts'], 'url.query_args');
-    array_push($build['table']['#cache']['tags'], 'apigee_my_subscriptions');
+      $subscriptions[SubscriptionInterface::STATUS_ENDED]['entities'][] = $entity;
+    }
+
+    foreach ($subscriptions as $key => $subscription) {
+      $build[$key]['heading'] = [
+        '#type' => 'html_tag',
+        '#value' => $subscription['title'],
+        '#tag' => 'h3',
+      ];
+
+      $build[$key]['table'] = [
+        '#type' => 'table',
+        '#header' => $this->buildHeader(),
+        '#title' => $subscription['title'],
+        '#rows' => [],
+        '#empty' => $this->t('There are no @label yet.', ['@label' => strtolower($this->entityType->getPluralLabel())]),
```

Comment:
This message doesn't really make a ton of sense when there are no results.  I wonder if instead of changing the message, whether it would be better to just not display the table at all *until* there is something to show?

<img width="1204" alt="Screen Shot 2019-06-07 at 2 14 11 PM" src="https://user-images.githubusercontent.com/60979/59124798-a0283000-892e-11e9-8f72-aa5d04f17b1f.png">

@cnovak Do you have any thoughts on this?

Reply:
Here is the whole page screenshot, for better context.

![purchased-plan-grouping](https://user-images.githubusercontent.com/60979/59125038-365c5600-892f-11e9-9b0e-0ed702b0ad45.png)

Reply:
FYI @earth2marsh The only problem with the table being gone is that I have found myself being confused before when things would disappear.  "Is it gone because I am looking in the wrong place, or because something is wrong w/the system?"

How about something like "No cancelled or expired plans"?

Reply:
I've updated the default message. (Note: this says subscriptions because it is using the configurable entity label, which is subscriptions by default)

![Purchased_plans___Drupal_8_m10n](https://user-images.githubusercontent.com/124599/59175736-abb26b80-8b67-11e9-9525-580eadca571a.jpg)